bazel/linux/defs.bzl

```diff
@@ -7,8 +7,51 @@ load("@bazel_skylib//lib:shell.bzl", "shell")
 
 def _kernel_tree_version(ctx):
     distro, version = ctx.attr.package.split("-", 1)
+    astore_path = ctx.path(ctx.attr.path.split("/")[-1])
```

Comment:
this is duplicating a lot of the code we have in astore_package rule.

Suggestion:
1) Add astore_package rule to enkit as is (just the astore_package).
2) Make the implementation a callable function. I would probably recommend doing something like here: https://github.com/enfabrica/enkit/blob/master/bazel/linux/runner.bzl#L33 (define the attributes needed in a dict, have a callable function that is not a rule). This way, we can add astore  download support to the kernel_image or ... kernel_whatever rules as well in 2-3 lines of code.
3) call the function from here, rather than duplicating the code.